src/callbacks_step/save_restart.jl

```diff
@@ -156,5 +160,32 @@ function load_restart_file(semi::AbstractSemidiscretization, restart_file)
     load_restart_file(mesh_equations_solver_cache(semi)..., restart_file)
 end
 
+"""
+    load_controller!(integrator, restart_file::AbstractString)
+
+Load the context information for PID- and PIControllers saved in a `restart_file`.
+"""
+function load_controller!(integrator, restart_file::AbstractString)
+    controller = integrator.opts.controller
+    # Read context information for controller
+    h5open(restart_file, "r") do file
+        # Ensure that necessary information was saved
+        if ("qold" in keys(attributes(file))) &&
+           ("dtpropose" in keys(attributes(file)))
```

Comment:
Instead of silently ignoring the case where the attributes are not present, I would do first a check on all properties that you expect, i.e., something like "if qold or dtpropose or err are missing, show an error 'missing data in restart file: ...'" If someone does not want to load these values, they can just not call the load function. But if they _do_ use this function, they should be able to expect that everything works